kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -51,33 +52,44 @@
 log = logging.getLogger('collector')
 
 
-class KubeletCheck(AgentCheck, CadvisorScraper):
+class KubeletCheck(CadvisorPrometheusScraperMixin, OpenMetricsBaseCheck, CadvisorScraper):
     """
     Collect metrics from Kubelet.
     """
 
     def __init__(self, name, init_config, agentConfig, instances=None):
-        super(KubeletCheck, self).__init__(name, init_config, agentConfig, instances)
-
         self.NAMESPACE = 'kubernetes'
-
         if instances is not None and len(instances) > 1:
             raise Exception('Kubelet check only supports one configured instance.')
         inst = instances[0] if instances else None
 
+        cadvisor_instance = self._create_cadvisor_prometheus_instance(inst)
+        kubelet_instance = self._create_kubelet_prometheus_instance(inst)
+        generic_instances = [cadvisor_instance, kubelet_instance]
+        super(KubeletCheck, self).__init__(name, init_config, agentConfig, generic_instances)
+
         self.cadvisor_legacy_port = inst.get('cadvisor_port', CADVISOR_DEFAULT_PORT)
         self.cadvisor_legacy_url = None
 
-        self.cadvisor_scraper = CadvisorPrometheusScraper(self)
-
-        self.kubelet_scraper = PrometheusScraper(self)
-        self.kubelet_scraper.NAMESPACE = 'kubernetes'
-        self.kubelet_scraper.metrics_mapper = {
-            'apiserver_client_certificate_expiration_seconds': 'apiserver.certificate.expiration',
-            'rest_client_requests_total': 'rest.client.requests',
-            'kubelet_runtime_operations': 'kubelet.runtime.operations',
-            'kubelet_runtime_operations_errors': 'kubelet.runtime.errors',
-        }
+        self.cadvisor_scraper_config = self.get_scraper_config(cadvisor_instance)
+        self.kubelet_scraper_config = self.get_scraper_config(kubelet_instance)
+
+    def _create_kubelet_prometheus_instance(self, instance):
+        kubelet_instance = deepcopy(instance)
+        kubelet_instance.update({
+            'namespace': self.NAMESPACE,
+            'prometheus_url': instance.get('kubelet_metrics_endpoint', 'dummy_url/kubelet'),
```

Comment:
Good point!

The reason why we need a dummy url is because the base class needs an endpoint to use as the key to the `config_map` it generates. The reason it has `kubelet` at the end of the endpoint is because we need to differentiate the key between the `kubelet` scraper and the `cadvisor` scraper.